src/lib/common/github/issuereporter.ts

```diff
@@ -0,0 +1,223 @@
+import * as Octokit from '@octokit/rest';
+import { IssuesUpdateParams } from '@octokit/rest';
+
+import { IssueData } from '../../types/issuedata';
+
+const { NODE_ENV, environment } = process.env; // eslint-disable-line no-process-env
+
+const production = NODE_ENV !== 'development';
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+export class IssueReporter {
+
+    /* eslint-disable no-process-env */
+    private GITHUB_API_TOKEN = process.env.GITHUB_API_TOKEN;
+    private GITHUB_OWNER = process.env.GITHUB_OWNER;
+    private GITHUB_REPO = process.env.GITHUB_REPO;
+    private GITHUB_DATA;
+    /* eslint-enable no-process-env */
+
+    private octokit;
+
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    public constructor() {
+        this.GITHUB_DATA = {
+            owner: this.GITHUB_OWNER,
+            repo: this.GITHUB_REPO
+        };
+        this.octokit = new Octokit({
+            baseUrl: 'https://api.github.com',
+            headers: {
+                accept: 'application/vnd.github.v3+json',
+                'user-agent': 'webhint'
+            },
+            timeout: 0
+        });
+
+        this.octokit.authenticate({
+            token: this.GITHUB_API_TOKEN,
+            type: 'oauth'
+        });
+    }
+
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    private addIssueComment(issue, issueData: IssueData) {
+        return this.octokit.issues.createComment({
+            body: this.getErrorMessage(issueData),
+            number: issue.issue_number,
```

Comment:
`number` is deprecated. Use `issue_number` instead.
```suggestion
            issue_number: issue.issue_number,
```